### PR TITLE
Gardener 1

### PR DESCRIPTION
**Pull request by gfr10598**
Add datatstore code for persisting some batch processing state.
This also adds some structs and comments that outline how this will be used, and code that tickles it in the 5 minute periodic loop.

Also does some code organization, and tweaks the healthcheck related code.

---
This change is [Reviewable](https://reviewable.io/reviews/m-lab/etl/442)

### CODE REVIEW

File: cmd/gardener/gardener.go [hunk 1]
```diff
@@ -5,41 +5,133 @@
...
-	mux.HandleFunc("/debug/pprof/", pprof.Index)
-	mux.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
-	mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
-	mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
-	mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
-	go http.ListenAndServe(":9090", mux)
+// ###############################################################################
+// All DataStore related code and variables
+// ###############################################################################
+const (
```
Comment (pboothe) [inline cmd/gardener/gardener.go]: Please use a namespace as well.

File: cmd/gardener/gardener.go [hunk 2]
```diff
@@ -61,6 +153,71 @@ func QueuerFromEnv() (batch.Queuer, error) {
...
+
+	err = Save(batchStateKey, &bs)
+	return bs, err
+}
+
+// ###############################################################################
+//  Top level service control code.
+// ###############################################################################
+
+// periodic will run approximately every 5 minutes.
```
Comment (pboothe) [inline cmd/gardener/gardener.go]: Do we want to add some randomness here?

**Comment by gfr10598** [comment]
Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---

*[cmd/gardener/gardener.go, line 28 at r1](https://beta.reviewable.io/reviews/m-lab/etl/442#-L5ALFoQWH6mhdIVFEta-r1-28:-L5AvZUY4BzkbQXA5Q_I:b-896fix) ([raw file](https://github.com/m-lab/etl/blob/85a7ad2189d6a3216c1a8a82249424015029bb39/cmd/gardener/gardener.go#L28)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

Please use a namespace as well.
</blockquote></details>

Done.

---

*[cmd/gardener/gardener.go, line 189 at r1](https://beta.reviewable.io/reviews/m-lab/etl/442#-L5ALFoQWH6mhdIVFEta-r1-189:-L5AvbDz4izqS6n_f15t:b-7isrph) ([raw file](https://github.com/m-lab/etl/blob/85a7ad2189d6a3216c1a8a82249424015029bb39/cmd/gardener/gardener.go#L189)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

Do we want to add some randomness here?
</blockquote></details>

There is already some implicit randomness based on the execution time.  There is no need for randomness, since this will be a singleton.  Adding comment.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/m-lab/etl/442)*

**Comment by gfr10598** [comment]
PTAL.

---

Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/m-lab/etl/442#-:-L5Avvh-2WKnZBcjIvFB:b-83330m)*

**Comment by pboothe** [comment]
:lgtm:

---

Reviewed 2 of 4 files at r1.
Review status: 2 of 4 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/etl/442#-:-L5Ayfa9eRRocmEVyObM:bnfp4nl)*

**Pull request merged**